sdks/java/core/src/main/java/org/apache/beam/sdk/util/FileIOChannelFactory.java

```diff
@@ -36,24 +36,38 @@
 import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
 import java.nio.file.PathMatcher;
-import java.nio.file.Paths;
 import java.util.Collection;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.regex.Matcher;
 
 /**
- * Implements IOChannelFactory for local files.
+ * Implements {@link IOChannelFactory} for local files.
  */
 public class FileIOChannelFactory implements IOChannelFactory {
   private static final Logger LOG = LoggerFactory.getLogger(FileIOChannelFactory.class);
 
+  /**
+   *  Converts the given file spec to a java {@link File}. If {@code spec} is actually a URI with
+   *  the {@code file://} prefix, then this function will ensure that the returned {@link File}
+   *  has the correct path.
+   */
+  private static File specToFile(String spec) {
+    // Strip the "file://" prefix from the spec, if present.
+    if (spec.toLowerCase().startsWith("file://")) {
+      return new File(spec.substring("file://".length()));
```

Comment:
Does this work on Windows? [A quick Google search](http://stackoverflow.com/questions/18520972/converting-java-file-url-to-file-path-platform-independent-including-u#comment27236348_18520972) turns up the following snippet: `Paths.get(spec).toFile()`